schema/string_test.go

```diff
@@ -29,9 +32,20 @@ func TestDecodeString_ErrorCheckingConstraints(t *testing.T) {
 		{"InvalidMaxLength_UUID", "6fa459ea-ee8a-3ca4-894e-db77e160355e", stringUUID, Constraints{MaxLength: 1}},
 		{"InvalidMaxLength_Email", "foo@bar.com", stringEmail, Constraints{MaxLength: 1}},
 		{"InvalidMaxLength_URI", "http://google.com", stringURI, Constraints{MaxLength: 1}},
+		{"InvalidPattern_UUID", "6fa459ea-ee8a-3ca4-894e-db77e160355e", stringUUID, Constraints{Pattern: "^[0-9a-f]{1}-.*"}},
+		{"InvalidPattern_Email", "foo@bar.com", stringEmail, Constraints{Pattern: "[0-9].*"}},
+		{"InvalidPattern_URI", "http://google.com", stringURI, Constraints{Pattern: "^//.*"}},
```

Comment:
Here you should test only the compiled version. The pattern to compiledPattern conversion should be done in the TestRead method.

Lets use the [MustCompile](https://golang.org/pkg/regexp/#MustCompile). Thus, a case could be:

```go
{"InvalidPattern_UUID", "6fa459ea-ee8a-3ca4-894e-db77e160355e", stringUUID, Constraints{compiledPattern: regexp.MustCompile("^[0-9a-f]{1}-.*")}}
```

Reply:
ok, The reason why I don't use `MustCompile` before is it will panic the program when it compile fails rather than issue an error.
I think we can control those testing pattern to make sure it always compile success, so, it's fine to use `MustCompile` to pass complied patten.